### PR TITLE
Add @kmer_str macro to allow kmer string literals

### PR DESCRIPTION
**Pull request by kdmurray91**
Also adds documentation and a doctest.

### CODE REVIEW

**Comment by Ward9250** [comment]
Can the doctest be added as an actual test?

**Comment by kdmurray91** [comment]
Should be good to go now @Ward9250

File: src/seq/kmer.jl [hunk 1]
```diff
@@ -273,3 +273,21 @@ function swap{T,k}(kmer::Kmer{T,k}, i, j)
     x = ((b >> i) $ (b >> j)) & UInt64(0x03)
     return Kmer{T,k}(b $ ((x << i) | (x << j)))
 end
+
+# String literal
+# --------------
+
+macro kmer_str(seq, flag)
+    if flag == "s"
```
Comment (bicycle1885) [inline src/seq/kmer.jl]: There is no need to have this flag. `Kmer` is always immutable and there is no distinction of 'static' or 'dynamic'.

**Comment by bicycle1885** [comment]
Can you mention this macro in NEWS.md?

File: src/seq/kmer.jl (hunk 1 above)
Comment (kdmurray91) [inline src/seq/kmer.jl]: @bicycle1885 so remove the whole macro @ lines 280-289?

Comment (bicycle1885) [inline src/seq/kmer.jl]: Yes.

**Comment by bicycle1885** [comment]
Thank you, @kdmurray91!

**Pull request merged**